internal/handlers/common/aggregations/operators/accumulators/accumulators.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package accumulators provides aggregation accumulator operators.
+package accumulators
+
+import (
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// newAccumulatorFunc is a type for a function that creates an accumulation operator.
+type newAccumulatorFunc func(expression *types.Document) (Accumulator, error)
+
+// Accumulator is a common interface for aggregation accumulation operators.
+type Accumulator interface {
+	// Accumulate documents and returns the result of applying operator.
+	Accumulate(in []*types.Document) (any, error)
+}
+
+// Get returns accumulator for provided value.
+func Get(value any, stage, key string) (Accumulator, error) {
```

Comment:
The order of parameter is a bit strange, I think it's more natural to have stage, key, value. What do you think? :thinking:

Reply:
Maybe `NewAccumulator` is a bit more specific for this function.